packages/expo-cli/src/commands/__tests__/upgrade-test.ts

```diff
@@ -0,0 +1,319 @@
+        main: 'index.js',
+      },
+      null,
+      2
+    );
+
+    const appJson = {
+      name: 'testing 123',
+      version: '0.1.0',
+      slug: 'testing-123',
+      sdkVersion: '33.0.0',
+    };
+
+    const expoPackageJson = JSON.stringify({
+      name: 'expo',
+      version: '35.0.0',
+    });
+
+    vol.fromJSON({
+      '/from-app-json-UNVERSIONED/package.json': packageJson,
+      '/from-app-json-UNVERSIONED/app.json': JSON.stringify({
+        expo: { ...appJson, sdkVersion: 'UNVERSIONED' },
+      }),
+      '/from-app-json-UNVERSIONED/app.config.js': `module.exports=({config}) => ({ ...config, sdkVersion: '34.0.0' })`,
+      '/from-app-json-UNVERSIONED/node_modules/expo/package.json': expoPackageJson,
+      '/from-app-json-UNVERSIONED/node_modules/expo/bundledNativeModules.json': JSON.stringify({}),
+
+      '/from-app-json/package.json': packageJson,
+      '/from-app-json/app.json': JSON.stringify({ expo: { ...appJson } }),
+      '/from-app-json/app.config.js': `module.exports=({config}) => ({ ...config, sdkVersion: '34.0.0' })`,
+      '/from-app-json/node_modules/expo/package.json': expoPackageJson,
+      '/from-app-json/node_modules/expo/bundledNativeModules.json': JSON.stringify({}),
+
+      '/from-app-config-js/package.json': packageJson,
+      '/from-app-config-js/app.json': JSON.stringify({
+        expo: { ...appJson, sdkVersion: undefined },
+      }),
+      '/from-app-config-js/app.config.js': `module.exports=({config}) => ({ ...config, sdkVersion: '34.0.0' })`,
+      '/from-app-config-js/node_modules/expo/package.json': expoPackageJson,
+      '/from-app-config-js/node_modules/expo/bundledNativeModules.json': JSON.stringify({}),
+
+      '/from-expo-package/package.json': packageJson,
+      '/from-expo-package/app.json': JSON.stringify({
+        expo: { ...appJson, sdkVersion: undefined },
+      }),
+      '/from-expo-package/app.config.js': `module.exports=({config}) => ({ ...config })`,
+      '/from-expo-package/node_modules/expo/package.json': expoPackageJson,
+      '/from-expo-package/node_modules/expo/bundledNativeModules.json': JSON.stringify({}),
+    });
+  });
+  afterAll(() => {
+    vol.reset();
+  });
+
+  // Ensure we delete the app.json sdkVersion if it's defined
+  it(`delete the app.json sdkVersion if it's defined`, async () => {
+    require('@expo/config').writeConfigJsonAsync = jest.fn((...props) => {
+      return jest.requireActual('@expo/config').writeConfigJsonAsync(...props);
+    });
+    const projectRoot = '/from-app-json';
+    await upgradeAsync(
+      { projectRoot, workflow: 'bare', requestedSdkVersion: null },
+      { yarn: true }
+    );
+    expect(require('@expo/config').writeConfigJsonAsync).toBeCalledTimes(1);
+    expect(require('@expo/config').writeConfigJsonAsync).toHaveBeenLastCalledWith(projectRoot, {
+      sdkVersion: undefined,
+    });
+
+    const { exp, rootConfig: json } = await require('@expo/config').readConfigJsonAsync(
```

Comment:
i think we should remove the part of this test where we check that `readConfigJsonAsync` gets the correct `sdkVersion` - in this test specifically intend to verify that we are removing `sdkVersion` from app.json and not that `readConfigJsonAsync` works as expected

Reply:
We want to ensure that after the test is run the resulting `sdkVersion` is now updated to an expected value. We are checking the raw results to ensure that the command has successfully "upgraded" the project.

Reply:
yeah i understand it just i think adds a lot of responsibility to a single unit test. i'm not a fanatic about these things so we can just leave it if you want to